views/view-logical-workers.php

```diff
@@ -36,57 +42,79 @@
     }
 }
 
-// Gathers VM node if a worker
-$personality = Array();
-
-$vms = get_aquilon_report("host_personality_branch_nubes");
-foreach ($vms as $name => $values) {
-    foreach ($values as $key => $value) {
-        if (strpos($value, "workernode") !== false) {
-            $personality[$name][$value] = $name;
-        }
-    }
-}
+$personalities = get_aquilon_report($CONFIG, 'host_personality_branch');
+$models = get_aquilon_report($CONFIG, 'report_host_model');
 
 // Generates main array
 $results = Array();
 $results['config'] = $config;
+
+
 foreach ($all_clusters as $name => $panels) {
-    if (($panels !== "vm-nubes") or (array_key_exists($name, $personality) === true)) {
+    if (($panels !== "vm-nubes") and (array_key_exists($name, $personalities) === true)) {
 
         $group = '';
         $panel = $panels;
         $cluster = '';
 
+        if (array_key_exists($name, $personalities)) {
+            $cluster = $personalities[$name]['personality'];
+        };
+
         $results[$group][$panel][$cluster][$name] = Array();
         if (array_key_exists($name, $all_notes)) {
             $results[$group][$panel][$cluster][$name]['note'] = $all_notes[$name];
-        }
+        };
         if (nagios($name) !== Null) {
             $results[$group][$panel][$cluster][$name]['nagios'] = nagios($name);
-        }
+        };
         if (array_key_exists($name, $all_status)) {
             $results[$group][$panel][$cluster][$name]['status'] = $all_status[$name];
             unset($all_status[$name]);
-        }
+        };
     }
 }
 
-// Render all remaining hosts from the batch state table
-// this catches any hosts from cloud-bursting or other dynamic sources
+
+// Render all leftover hosts from the batch state table
+// this catches any hosts created via cloud-bursting
 foreach ($all_status as $hostname => $status) {
     if ($hostname !== '') {
         $source = '';
         if (sizeof($status) > 0) {
             $source = array_values($status);
             $source = $source[0];
         }
-        $results['']['dynamic'][$source][$hostname] = Array(
+
+        $model = $source;
+        if (array_key_exists($hostname, $models)) {
+            $model = $models[$hostname];
+        };
+
+        $personality = '∅';
+        if (array_key_exists($hostname, $personalities)) {
+            if (array_key_exists('personality', $personalities[$hostname])) {
+                $personality = $personalities[$hostname]['personality'];
+            };
+        };
+
+        $details = Array(
             'status' => $status,
         );
+
+        if (array_key_exists($hostname, $all_notes)) {
+            $details['note'] = $all_notes[$hostname];
+        };
+        if (nagios($hostname) !== Null) {
```

Comment:
Should the references to Nagios be removed?

Reply:
Not in this PR.